docs/types/reference-types.rst

```diff
@@ -192,14 +192,18 @@ Array Literals
 ^^^^^^^^^^^^^^
 
 An array literal is a comma-separated list of one or more expressions, enclosed
-in square brackets (``[...]``). For example ``[1, a, f(3)]``. There must be a
-common type all elements can be implicitly converted to. This is the elementary
-type of the array.
+in square brackets (``[...]``). For example ``[1, a, f(3)]``. The type of the
+array literal is determined as follows:
 
-Array literals are always statically-sized memory arrays.
+It is always a statically-sized memory array with a length being equal to the
+number of expressions.
+
+The base type of the array is the type of one of the expressions such that all
+other expressions can be implicitly converted to it. It is a type error
```

Comment:
What if there are two types where all the other expressions can be converted to either? Which one is chosen?

Reply:
The first. Will clarify.